Feat/add fetch

thadjbullet opened the pull request:
Не сделано:
- Пагинация
- Фильтр по размерам
- Анимация

Короче то, что успел за 2 дня за компом. Костыльно, понимаю, но конца недели допилю канеш, хотя тебе уже не важно)
В остальном все сделано (даже вроде по заданию:)

Commit:
add: fetch to List & Item

fix: some issues

Commit:
add: Get method, price & title for Item

fix: BuyButton & BigImage

Commit:
fix: Sale label

Commit:
fix: readme

imevro commented on README.md:
```diff
@@ -2,7 +2,7 @@
 Clone or download repository
 
-Install dependsies
+Install dependcies
```
covfefe

thadjbullet replied:
Че это?

imevro replied:
https://www.google.ru/search?q=dependcies&oq=dependcies&aqs=chrome..69i57j0l5.310j0j7&sourceid=chrome&ie=UTF-8

imevro commented on src/Api.jsx:
```diff
@@ -0,0 +1,12 @@
+    method: 'GET',
+    headers: {
+      'Content-Type': 'text/plain',
+      Accept: '*/*',
```
Хуйня

imevro commented on src/Api.jsx:
```diff
@@ -0,0 +1,12 @@
+const baseUrl = 'https://erodionov-adidas-fake-api.now.sh/v1';
```
Почемк файл называется Api.jsx? Это что, компонент?

imevro commented on src/Products/Item/Gallery/index.jsx:
```diff
@@ -2,24 +2,27 @@
-const Container = styled.div`
-  display: flex;
-`;
+import GetImageLink from '../../../components/GetImageLink';
```
Почему компонент называется как функция?

imevro commented on src/Products/Item/Gallery/index.jsx:
```diff
@@ -57,25 +48,37 @@ export default class GalleryItem extends React.Component {
 
   render() {
+    const item = this.props.item;
+    const images = this.props.item.images;
```
Дестракт

imevro commented on src/Products/Item/Gallery/index.jsx:
```diff
@@ -57,25 +48,37 @@ export default class GalleryItem extends React.Component {
+        <BigImage
+          src={
+            images
+              ? GetImageLink(
```
Вынеси эту проверку в компонент

imevro replied:
Или функцию что у тебя там-то

imevro commented on src/Products/Item/index.jsx:
```diff
@@ -34,8 +35,31 @@ const Container = styled.div`
+
+  fetchImages(props) {
+    const url = props.match.url;
+    Get(url)
```
Get это компонент?

imevro commented on src/Products/Item/index.jsx:
```diff
@@ -34,8 +35,31 @@ const Container = styled.div`
+  fetchImages(props) {
+    const url = props.match.url;
+    Get(url)
+      .then(json => ({
```
Вынеси в функцию transformInputValues например

imevro commented on src/Products/Item/index.jsx:
```diff
@@ -34,8 +35,31 @@ const Container = styled.div`
   constructor(props) {
     super(props);
-    this.state = { color: '#c5c5c5' };
+    this.state = { color: '#c5c5c5', item: {} };
```
Хуитем

imevro commented on src/Products/List/Card.jsx:
```diff
@@ -5,6 +5,7 @@ import styled from 'styled-components';
 import { NavLink } from 'react-router-dom';
 
 import Label from '../../components/Label';
+import formatPrice from '../../components/FormatPrice';
```
что с неймингом-то

imevro replied:
почему у тебя функция лежит в components и называется в паскалькейсе?

imevro commented on src/Products/List/Card.jsx:
```diff
@@ -45,10 +46,12 @@ const SaleLabel = styled(Label)`
-    {props.isSale && <SaleLabel>Sale</SaleLabel>}
+    <Img src={image} />
+    <Link to={to} activeClassName={isSale ? 'active' : ''}>
+      {formatPrice(cost, '$')}
```
доллар есть в ответе сервера под currency, нахуй ты хардкодишь

imevro commented on src/Products/List/Filter/FilterSize.jsx:
```diff
@@ -19,30 +20,33 @@ const Span = styled.span`
+export default class FilterSize extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { selectedId: 0 };
```
Айди или индекс

imevro commented on src/Products/List/Filter/FilterSize.jsx:
```diff
@@ -19,30 +20,33 @@ const Span = styled.span`
 
-  &:hover {
-    color: #4d42f8;
+  selectButton(selectedId) {
```
Императивный нейминг — говно

imevro commented on src/Products/List/Filter/SizeButton.jsx:
```diff
@@ -0,0 +1,40 @@
+    this.handleOnClick = this.handleOnClick.bind(this);
+  }
+
+  handleOnClick() {
```
onClick либо handleClick, но не handleOnClick

imevro commented on src/Products/List/index.jsx:
```diff
@@ -1,11 +1,13 @@
-/* eslint-disable global-require */
+/* eslint-disable global-require, quote-props */
```
Убирай нахуй

imevro commented on src/Products/List/index.jsx:
```diff
@@ -17,30 +19,55 @@ const Content = styled.div`
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: [],
```
хуитемс

imevro commented on src/Products/List/index.jsx:
```diff
@@ -17,30 +19,55 @@ const Content = styled.div`
+
+    Get(url)
+      .then(json => this.setState({ items: json.items }))
+      .catch(error => global.console.log('request failed', error));
```
какой глобал вообще

thadjbullet replied:
Eslint надо спрашивать

imevro commented on src/components/FormatPrice.jsx:
```diff
@@ -0,0 +1,2 @@
+export default (price, currency) =>
+  (price !== undefined ? `${currency}${(price / 100).toFixed(2)}` : '$0.00');
```
вообще не возвращай нихуя чем цену в нуле

Commit:
fix: all remarks

Commit:
fix: rename api.jsx -> api.js
disable eslint no-console warning

Commit:
fix: change formatPrice & imageLInk extension to .js
add: transfornInputValue to Products/List
Little refactor

Commit:
fix: SizeButton select

Commit:
del: delete unsed images

imevro commented on src/Products/Item/Gallery/BigImage.jsx:
```diff
@@ -0,0 +1,26 @@
+export default ({ item, index }) => (
+  <BigImage
+    src={
+      item.images
```
убирай эти вычисления в пропах

imevro replied:
внутри компонента ебашь проверки

thadjbullet replied:
Это и есть компонент. Я учел прошлое замечание и вынес его отдельно.

imevro commented on src/Products/Item/Gallery/index.jsx:
```diff
@@ -48,31 +38,18 @@ export default class GalleryItem extends React.Component {
         <Gallery>
-          {images
-            ? images.map((image, index) => (
+          {this.props.item.images
```
хуевая проверка, лучше нулевой массив по-умолчанию держи

imevro commented on src/Products/Item/index.jsx:
```diff
@@ -38,29 +38,44 @@ export default class Page extends React.Component {
   }
 
+  /* eslint-disable class-methods-use-this */
+  transformInputValues(json) {
```
зачем это в классе? Вынеси в функцию. Тебе еслинт об этом даже говорит

imevro commented on src/Products/Item/index.jsx:
```diff
@@ -38,29 +38,44 @@ export default class Page extends React.Component {
+        res
+          .json()
+          .then(json =>
+            this.setState({ item: this.transformInputValues(json) }),
```
ХУИТЕМ

Переименуй в продукт, камон

imevro commented on src/Products/List/index.jsx:
```diff
@@ -23,44 +21,66 @@ export default class ProductsList extends React.Component {
-      .then(json => this.setState({ items: json.items }))
-      .catch(error => global.console.log('request failed', error));
+  /* eslint-disable class-methods-use-this */
+  transformInputValues(json) {
```
у тебя повторяется, вынеси в отдельный хелпер

imevro commented on src/Products/List/index.jsx:
```diff
@@ -23,44 +21,66 @@ export default class ProductsList extends React.Component {
-        <Filter sizes={items.length !== 0 ? items[0].sizes : null} />
+        <Filter
+          sizes={
+            this.state.products.length !== 0
```
убрать вычисления из пропа

imevro commented on src/api.js:
```diff
@@ -0,0 +1,3 @@
+const baseUrl = 'https://erodionov-adidas-fake-api.now.sh/v1';
+
+export default url => fetch(baseUrl + url);
```
пост, пут, делит ты как хуярить будешь

imevro commented on src/formatPrice.js:
```diff
@@ -0,0 +1,12 @@
+export default ({ price, currency }) => {
+  switch (currency) {
```
Не свитч, а объект

imevro commented on src/formatPrice.js:
```diff
@@ -0,0 +1,12 @@
+    case 'EURO':
+      return `€${price}`;
+    case 'RUB':
+      return `₽${price}`;
```
читай стандарты форматирования валют

imevro replied:
₽500 долбоебизм

Commit:
fix: formatPrice, List/index
rename item -> product

Commit:
del: search
minor fixes

Commit:
fix: remove formatPrice & imageLink to utilities
minor fixes
add: return search back

Commit:
refactor: remove transformInputValue from components

Commit:
add: fetchSizes

Commit:
add: loading component, working filter

Commit:
add: copyright page

Commit:
fix: sort sizes

Commit:
fix: import method
add: post, put, delete to api

Commit:
fix: copyright line lenght

imevro commented on src/Products/Item/Gallery/index.jsx:
```diff
@@ -59,23 +40,24 @@ export default class GalleryItem extends React.Component {
-        </Container>
+        <BigImage
+          product={this.props.product}
+          index={this.state.selectedIndex}
```
index={selectedIndex}

thadjbullet replied:
Здесь я не понял, что ты хотел сказать.

imevro replied:
консистентность где

imevro commented on src/Products/Item/Gallery/index.jsx:
```diff
@@ -59,23 +40,24 @@ export default class GalleryItem extends React.Component {
-              onChangeImage={() => this.handleChangeImage(index)}
-            />
-          ))}
+          {this.props.product.images &&
```
зачем, если можно products.images по-умолчанию сделать пустым массивом

imevro replied:
тогда эта проверка не будет нужна и .map будет проходиться (не проходиться) по пустому массиву

imevro commented on src/Products/Item/index.jsx:
```diff
@@ -47,16 +71,22 @@ export default class Page extends React.Component {
-          <Gallery />
-        </Item>
+        {this.state.product
+          ? <Product>
```
Тернарники выглядят как говно

imevro commented on src/Products/List/Filter/FilterSize.jsx:
```diff
@@ -19,30 +20,40 @@ const Span = styled.span`
 
-  &:hover {
-    color: #4d42f8;
+  isSelected(id) {
```
Почему у тебя функция-обработчик-эвента называется как булеан свойство? втф?

imevro commented on src/Products/List/index.jsx:
```diff
@@ -15,32 +16,86 @@ const Container = styled.main`
+    this.setState({ filter: null });
+  }
+
+  handleFilter(filter) {
```
хэндлфильтр что

imevro commented on src/Products/List/index.jsx:
```diff
@@ -15,32 +16,86 @@ const Container = styled.main`
+
+  filterProducts(products) {
+    return this.state.filter
+      ? products.filter(item => item.sizes.includes(this.state.filter))
```
тернарники пиздец

imevro replied:
если у тебя будет нормальное исходное значение, то тебе не нужен будет тернарник

thadjbullet replied:
Так тут проверка на наличие выделенного размера и фильтрация результатов. Но я переделал на if.

imevro commented on src/Products/List/index.jsx:
```diff
@@ -15,32 +16,86 @@ const Container = styled.main`
+  fetchData(props) {
+    get(props)
+      .then(res =>
+        res.json().then(json =>
```
дестракт используй

imevro replied:
почему res.json() не в api? нравится каждый раз писать?

Commit:
del: Loading component
fix: isSelected -> selectId
handleFilter -> handleChooseFilter
use destract in fetchData
remove * ? * : *
move json() to api.js

Commit:
add: bettercode analyze

Commit:
fix: move bettercode to project root

Commit:
fix: minor fix

Commit:
Merge remote-tracking branch 'origin/feat/add-fetch' into feat/add-fetch

Commit:
fix: move back bettercode.yml :)

Commit:
test

Commit:
test

imevro reviewed (changes requested):
фикси оставшиеся комментарии, зачем пр прислал

Commit:
fix: change initial filter state to ''

thadjbullet commented:
Убрал все тернарники.

Commit:
fix: move utilities.js to Product folder
